ReactWindows/ReactNative.Shared/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,229 @@
+using ReactNative.UIManager;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Automation.Peers;
+
+namespace ReactNative.Common
+{
+    /// <summary>
+    /// Accessibility helper methods.
+    /// </summary>
+    public static class AccessibilityHelper
+    {
+        #region ImportantForAccessibility
+
+        /// <summary>
+        /// Sets the ImportantForAccessibility property for uiElement.
+        /// It uses AutomationProperties.AccessibilityView to implement the element and its children
+        /// visibility to narrator. ImportantForAccessibility value is stored as an attached property.
+        /// </summary>
+        /// <param name="uiElement"></param>
+        /// <param name="importantForAccessibilityNew"></param>
+        public static void SetImportantForAccessibility(UIElement uiElement, ImportantForAccessibility importantForAccessibilityNew)
+        {
+            var importantForAccessibilityOld = GetImportantForAccessibilityAttached(uiElement);
+            if (importantForAccessibilityOld == importantForAccessibilityNew)
+            {
+                return;
+            }
+
+            SetImportantForAccessibilityAttached(uiElement, importantForAccessibilityNew);
+
+            var uiElementAutomationPeer = FrameworkElementAutomationPeer.FromElement(uiElement);
+            // If this element is 'hidden' by a parrent, do not update AccessibilityView property.
+            if (IsHiddenElement(uiElementAutomationPeer))
+            {
+                return;
+            }
+
+            // Update AccessibilityView property accordingly.
+            switch (importantForAccessibilityNew)
+            {
+                case ImportantForAccessibility.Auto:
```

Comment:
>Auto [](start = 47, length = 4)

Suppose we set ImportantForAccessibility to `No` and then to `Auto`. Now that ImportantForAccessibility is `Auto`, how does `AutomationProperties.SetAccessibilityView` get back to its default state?

Reply:
Setting ImportantForAccessibility to 'No' will set AcceessibilityView to "Raw" for the element but will not affect the childrent. Setting it back to "Auto" will delete the "AccessibilityView" property, effectively returning it to whatever is the default for the control. The children will not be changed.

---
In reply to: [163133728](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163133728) [](ancestors = 163133728)